Updated progress topic (#408)
IgniteUI/indigo-design-docfx: en/components/progress.md

<img class="responsive-img" src="../images/progress_demo.png" srcset="../images/progress_demo@2x.png 2x" />

## Shape and Type

> The Progress supports two layout types to fit the variety of use cases and layout requirements: a Circular Bar and a Linear Bar. Each of these two types has an Indeterminate type.

Reviewer: ```suggestion
The Progress supports two shape variants to fit the variety of use cases and layout requirements: a Circular Bar and a Linear Bar. Each of these offers a choice between two types Determinate and Indeterminate.
```

> ## Type

Reviewer: ```suggestion
## Shape and Type
```

> The Progress supports two layout types to fit the variety of use cases and layout requirements: a Circular Bar and a Linear Bar. Each of these two types has an Indeterminate type.

Reviewer: ```suggestion
The Progress supports two layout shapes to fit the variety of use cases and layout requirements: a Circular Bar and a Linear Bar. Each of these layouts comes with an additional selection of the type of progression: Determinate and Indeterminate.
```

<img class="responsive-img" src="../images/progress_circular-progress-amount.png" srcset="../images/progress_circular-progress-amount@2x.png 2x" />
<img class="responsive-img" src="../images/progress_linear-progress-amount.png" srcset="../images/progress_linear-progress-amount@2x.png 2x" />

## Stripes

When using a Linear Bar layout shape for the Progress component, it will come with an additional option for Stripes over the Fill area. To remove the Stripes, simply set the `🌈 Stripes` color override to `transparent`.

<img class="responsive-img" src="../images/progress_linearstripes.png" srcset="../images/progress_linearstripes@2x.png 2x" />

## Styling

The Progress comes with styling flexibility through the various overrides for its text, as well as changing the colors of progress, base, stripes, fill and track where applicable.

<img class="responsive-img" src="../images/progress_striped.png" srcset="../images/progress_striped@2x.png 2x" />
<img class="responsive-img" src="../images/progress_clear.png" srcset="../images/progress_clear@2x.png 2x" />
<img class="responsive-img" src="../images/progress_twocolor.png" srcset="../images/progress_twocolor@2x.png 2x" />

## Usage

> In a Circular Bar, always use the actual value for the text label and, when adding more elaborate text in a Linear Bar, provide your best estimate for the state of completion of the task in time/files left, rather than displaying a generic string to the user. However, you may use the Indetermiante variants for both bars to display a Progress without certain and clear competion time.

Reviewer 2: ```suggestion
In a Circular Bar, always use the actual value for the text label and, when adding more elaborate text in a Linear Bar, provide your best estimate for the state of completion of the task in time/files left, rather than displaying a generic string to the user. However, you may use the Indeterminate variants for both bars to display a Progress without certain and clear completion time.
```

The Progress can be used in one of the following preset color combinations:

> - **default**

Reviewer: ```suggestion
- **default**: utilizing the base primary color by default
```

- error: utilizing the `error` theme color to show the progress
- info: utilizing the `info` theme color to show the progress
- success: utilizing the `success` theme color to show the progress
- warn: utilizing the `warn` theme color to show the progress

<img class="responsive-img" src="../images/progress_default.png" srcset="../images/progress_default@2x.png 2x" />
<img class="responsive-img" src="../images/progress_error.png" srcset="../images/progress_error@2x.png 2x" />
<img class="responsive-img" src="../images/progress_info.png" srcset="../images/progress_info@2x.png 2x" />
<img class="responsive-img" src="../images/progress_success.png" srcset="../images/progress_success@2x.png 2x" />
<img class="responsive-img" src="../images/progress_warn.png" srcset="../images/progress_warn@2x.png 2x" />

## Progress Amount

> The Progress allows you to choose the progress amount between six percentage variants. But you don't have this option for the Indeterminate Circular and Indeterminate Linear Bar types.

Reviewer: ```suggestion
When using a Determinate type of Progress, you may choose the Progress Amount between six percentage variants.
```

<img class="responsive-img" src="../images/progress_circular-progress-amount.png" srcset="../images/progress_circular-progress-amount@2x.png 2x" />

> <img class="responsive-img" src="../images/progress_linear-progress-amount.png" srcset="../images/progress_linear-progress-amount@2x.png 2x" />

Reviewer: ```suggestion
<img class="responsive-img" src="../images/progress_linear-progress-amount.png" srcset="../images/progress_linear-progress-amount@2x.png 2x" />

## Stripes

When using a Linear Bar layout shape for the Progress component, it will come with an additional option for Stripes over the Fill area. To remove the Stripes, simply set the `🌈 Stripes` color override to `transparent`.

@YoannaIvanova add an image instead of this text
```

| ---------------------------------------------------------------------------------- | -------------------------------------------------------------------------------------- |
| <img class="responsive-img" src="../images/progress_do1.png" srcset="../images/progress_do1@2x.png 2x" /> | <img class="responsive-img" src="../images/progress_dont1.png" srcset="../images/progress_dont1@2x.png 2x" /> |
| <img class="responsive-img" src="../images/progress_do2.png" srcset="../images/progress_do2@2x.png 2x" /> | <img class="responsive-img" src="../images/progress_dont2.png" srcset="../images/progress_dont2@2x.png 2x" /> |

> | <img class="responsive-img" src="../images/progress_do3.png" srcset="../images/progress_do3@2x.png 2x" /> | <img class="responsive-img" src="../images/progress_dont3.png" srcset="../images/progress_dont3@2x.png 2x" /> |

Reviewer: Can you only remove the stripes from the left image and I will approve
